pkg/models/spending_funding.go

```diff
@@ -0,0 +1,84 @@
+package models
+
+import (
+	"time"
+
+	"github.com/monetr/monetr/pkg/util"
+)
+
+type SpendingFunding struct {
+	tableName string `pg:"spending_funding"`
+
+	SpendingFundingId      uint64           `json:"spendingFundingId" pg:"spending_funding_id,notnull,pk,type:'bigserial'"`
+	AccountId              uint64           `json:"-" pg:"account_id,notnull,pk,on_delete:CASCADE,type:'bigint'"`
+	Account                *Account         `json:"-" pg:"rel:has-one"`
+	BankAccountId          uint64           `json:"bankAccountId" pg:"bank_account_id,notnull,pk,unique:per_bank,on_delete:CASCADE,type:'bigint'"`
+	BankAccount            *BankAccount     `json:"bankAccount,omitempty" pg:"rel:has-one" swaggerignore:"true"`
+	SpendingId             uint64           `json:"spendingId" pg:"spending_id,notnull,on_delete:CASCADE,type:'bigint'"`
+	Spending               *Spending        `json:"-" pg:"rel:has-one"`
+	FundingScheduleId      uint64           `json:"fundingScheduleId" pg:"funding_schedule_id,notnull,on_delete:CASCADE,type:'bigint'"`
+	FundingSchedule        *FundingSchedule `json:"-" pg:"rel:has-one"`
+	NextContributionAmount int64            `json:"nextContributionAmount" pg:"next_contribution_amount,notnull,use_zero"`
+}
+
+type SpendingFundingHelper []SpendingFunding
+
+func NewSpendingFundingHelper(funding []SpendingFunding) SpendingFundingHelper {
+	if len(funding) == 0 {
+		panic("must provide at least a single funding schedule")
+	}
+	return funding
+}
+
+func (s SpendingFundingHelper) GetNextTwoContributionDatesAfter(start time.Time, timezone *time.Location) (time.Time, time.Time) {
+	var first, second time.Time
+	for _, schedule := range s {
+		next := schedule.FundingSchedule.GetNextContributionDateAfter(start, timezone)
+		if next.Before(first) || first.IsZero() {
+			first = next
+		}
+	}
+
+	for _, schedule := range s {
+		next := schedule.FundingSchedule.GetNextContributionDateAfter(first, timezone)
+		if next.Before(second) || second.IsZero() {
+			second = next
+		}
+	}
+
+	return first, second
+}
```

Comment:
Write a test where two funding schedules would overlap. This should account for that but I want to be absolutely sure.